jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -431,6 +429,17 @@ public Response doResponseOk(final List<UUID> entitlementIds) {
         return callCompletionCreation.withSynchronization(callback, timeoutSec, callCompletion, callContext);
     }
 
+    private DateTime getDateTimeFromInput(final String inputDate, final TimeAwareContext timeAwareContext) {
+        if (inputDate == null) {
+            return null;
+        }
+        if (isDateTime(inputDate)) {
+            return toDateTime(inputDate);
+        } else {
+            return timeAwareContext.toUTCDateTime(toLocalDate(inputDate));
```

Comment:
Wondering if there is a regression here: if `inputDate` is `empty`, `isDateTime` is `false` and `toLocalDate` returns `null`. The `null` check was done prior to calling `toUTCDateTime` before. Maybe `toUTCDateTime` supports handling `null` values?

Reply:
Good catch! I've now fixed this [here](https://github.com/reshmabidikar/killbill/blob/5ded768eaf13082d107eec92d3fdc7c7ed716392/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java#L988). LMK if you still spot any issues.